monai/transforms/compose.py

```diff
@@ -158,7 +158,7 @@ def flatten(self):
         """
         new_transforms = []
         for t in self.transforms:
-            if isinstance(t, Compose) and not isinstance(t, OneOf):
+            if isinstance(t, Compose) and not isinstance(t, OneOf) and not isinstance(t, RandomOrder):
```

Comment:
Seems this line will cause a length error since it treats `RandomOrder` as an entire object. cc @wyli

Reply:
yeah, ideally the `Compose` as a base class shouldn't have the logic checking its subclasses. I think `OneOf` and `RandomOrder` should implement `flatten()` respectively instead of modifying `Compose.flatten()` directly. please help revise this if you know what I mean. otherwise I can do it later